File: 240328/top k 숫자/top-k-elements.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int k = sc.nextInt();

        TreeSet<Integer> treeSet = new TreeSet();

        for (int i = 0; i < n; i++) {
            treeSet.add(sc.nextInt());
        }

        Set<Integer> sortedSet = treeSet.descendingSet();
        List<Integer> list =  new ArrayList<>(sortedSet);
        StringBuilder answer = new StringBuilder();
        for (int i = 0; i < k; i++) {
            answer.append(list.get(i)).append(" ");
        }
        System.out.print(answer);
    }
}

File: 240328/top k 숫자/top-k-elements.kt
import java.util.*

val sc = Scanner(System.`in`)
val n = sc.nextInt()
val k = sc.nextInt()
val treeSet: TreeSet<Int> = TreeSet()

fun main() {
    for (i in 1..n) {
        treeSet.add(sc.nextInt())
    }
    
    val sortedSet = treeSet.descendingSet()
    val result = sortedSet.toList()
    val answer = StringBuilder()

    for (i in 0..k-1) {
        answer.append(result[i]).append(" ")
    }
    print(answer)
}